lib/mongodb.js

```diff
@@ -325,6 +325,15 @@ MongoDB.prototype.connect = function(callback) {
       if (callback) callback(err);
     }
 
+    const urlObj = new URL(self.settings.url);
+
+    if (urlObj.pathname !== '' &&
+        urlObj.pathname.split('/')[1] === '' &&
+        typeof self.settings.database === 'string') {
```

Comment:
@achrinza if `mongodb://localhost:27017` is used in `settings.url` the `pathname` will be `''` and in this case we should check if `settings.database` is set and set the database name as path name

```suggestion
    if ((urlObj.pathname === '' ||
        urlObj.pathname.split('/')[1] === '') &&
        typeof self.settings.database === 'string') {
```

Reply:
Ah, Iooks like I initially misunderstood when the default db name is injected by mongo driver. Thanks :+1: